src/plugins/intel_cpu/src/nodes/matrix_nms.cpp

```diff
@@ -376,7 +376,7 @@ void MatrixNms::execute(dnnl::stream strm) {
 
     // NMS-alike nodes are always transformed to NMSIEInternal node in case of legacy api, for compatibility.
     // And on the other hand in case of api 2.0, keep them internal dynamic for better performance and functionality.
-    if (isDynamicNode()) {
+    if (!m_outStaticShape) {
```

Comment:
Could you please elaborate on why I decided to change the condition?

Reply:
@ceciliapeng2011  could you help to explain it, thanks.
@maxnick  this condition change is from cecilia, I just submit it with the coverity issue of multiclass

Reply:
@maxnick It is a leftover of my last PR. https://github.com/openvinotoolkit/openvino/pull/16128#discussion_r1148861999 It's got merged before I fix it. that's all.

Reply:
Thank you, now it is clear.